base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
The `@nameable` instance variable should have both the getter and setter method available. To achieve the same, you can use `attr_accessor`. Kindly follow-up on this requirement.